dev/com.ibm.ws.jaxrs.defaultexceptionmapper/src/com/ibm/ws/jaxrs/defaultexceptionmapper/DefaultExceptionMapper.java

```diff
@@ -0,0 +1,63 @@
+/*******************************************************************************
+ * Copyright (c) 2020 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *******************************************************************************/
+
+package com.ibm.ws.jaxrs.defaultexceptionmapper;
+
+import com.ibm.websphere.ras.Tr;
+import com.ibm.websphere.ras.TraceComponent;
+import com.ibm.websphere.ras.annotation.TraceObjectField;
+import com.ibm.websphere.ras.annotation.TraceOptions;
+import com.ibm.ws.ras.instrument.annotation.InjectedFFDC;import java.lang.Override;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import javax.ws.rs.WebApplicationException;
+import javax.ws.rs.container.ResourceInfo;
+import javax.ws.rs.core.Context;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.ext.ExceptionMapper;
+import javax.ws.rs.ext.Provider;
+
+@Provider 
+public class DefaultExceptionMapper implements ExceptionMapper<Throwable> {
+   static final long serialVersionUID = 9083611544695926229L;
+   
+   @Context
+   private ResourceInfo resourceInfo;
+   
+   private final List<DefaultExceptionMapperCallback> callbacks = new LinkedList<>();
+   
+   DefaultExceptionMapper(Collection<DefaultExceptionMapperCallback> callbacks) {
+     this.callbacks.addAll(callbacks);
+   }
+   
+   @Override
+   public Response toResponse(Throwable t) {
+     Response response;
+     if (t instanceof WebApplicationException && ((WebApplicationException)t).getResponse() != null) {
+       response = ((WebApplicationException)t).getResponse();
+     } else {
+       response = Response.serverError().build();
+     } 
+     Map<String, Object> callbackHeaders = new HashMap<>();
+     for (DefaultExceptionMapperCallback callback : this.callbacks)
+       callbackHeaders.putAll(callback.onDefaultMappedException(t, response.getStatus(), this.resourceInfo)); 
```

Comment:
If we want callback implementations to be able to return null to indicate that they don't need to add any headers, then there should be a null check here to only call `putAll` if the returned map is not null.